### PR TITLE
updating makefile to throw an error if Earthly not installed

### PR DESCRIPTION
**Pull request by aronchick**

### CODE REVIEW

**Review by wdbaruni — COMMENTED** [review]
You have unrelated changes in `circleci/config.yml`. Are they intentional?

**Comment by aronchick** [comment]
> You have unrelated changes in `circleci/config.yml`. Are they intentional?

Just the one - it was a bad structure previously (it was assuming that you were on a branch for asdf). I was seeing the same problems here.

File: .circleci/config.yml [hunk 1]
```diff
@@ -155,14 +155,17 @@ commands:
...
           command: |
+            git config advice.detachedHead false
             if [ ! -d $HOME/.asdf ]; then
               git clone https://github.com/asdf-vm/asdf.git $HOME/.asdf --branch << parameters.asdf_version >>
               cd $HOME/.asdf
             else
               cd $HOME/.asdf
               git fetch origin
-              git pull
+              git fetch --all
```
Comment (wdbaruni) [inline .circleci/config.yml:165]: would `git fetch --tags` be enough here?

File: .circleci/config.yml [hunk 2]
```diff
@@ -155,14 +155,17 @@ commands:
...
             fi
+            git checkout master
+            git pull --all
```
Comment (wdbaruni) [inline .circleci/config.yml:168]: are these necessary?

File: .circleci/config.yml [hunk 3]
```diff
@@ -277,6 +281,8 @@ jobs:
       - run:
           name: Build tarball
           command: |
+            source "${BASH_ENV}"
+            cat "${BASH_ENV}"
```
Comment (wdbaruni) [inline .circleci/config.yml:285]: do we need to cat the envs?

**Review by wdbaruni — APPROVED** [review]
Minor comments. Can be merged once addressed

File: .circleci/config.yml (hunk 1 above)
Comment (aronchick) [reply .circleci/config.yml:165]: done

File: .circleci/config.yml (hunk 2 above)
Comment (aronchick) [reply .circleci/config.yml:168]: Yes. When pulling from asdf with a version, for some reason it puts you headless. I needed to checkout master, and then wanted to confirm everything was up to date (particularly just to ensure breaking caches).

File: .circleci/config.yml (hunk 3 above)
Comment (aronchick) [reply .circleci/config.yml:285]: this was for debugging - you're right, don't need to.

**Pull request closed without merging**